search-commons/src/main/java/no/unit/nva/search/resource/ContributorNodeFilter.java

```diff
@@ -0,0 +1,61 @@
+package no.unit.nva.search.resource;
+
+import static nva.commons.core.attempt.Try.attempt;
+
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+import no.unit.nva.constants.Words;
+import no.unit.nva.search.common.records.Midas;
+
+import nva.commons.core.JacocoGenerated;
+
+public class ContributorNodeFilter implements Midas {
+
+    @JacocoGenerated
+    ContributorNodeFilter() {}
+
+    public static ContributorNodeFilter verifiedOrNorwegian() {
+        return new ContributorNodeFilter();
+    }
+
+    @Override
+    public JsonNode transform(JsonNode source) {
+        var pointer = JsonPointer.compile(Words.ENTITY_DESCRIPTION_CONTRIBUTORS_PATH);
+        var target = source.at(pointer);
+        var elements = target.elements();
+        while (elements.hasNext()) {
+            var element = elements.next();
+            if (deleteAfter(element, 5)
+                    && isNotVerified(element)
+                    && hasNoNorwegianBasedContributor(element)) {
+                elements.remove();
+            }
+        }
+        return source;
+    }
+
+    private Boolean isNotVerified(JsonNode element) {
+        return attempt(
+                        () ->
+                                !Words.VERIFIED.equals(
+                                        element.get(Words.IDENTITY)
+                                                .get(Words.VERIFICATION_STATUS)
+                                                .asText()))
+                .orElse((e) -> Boolean.TRUE);
+    }
+
+    private Boolean hasNoNorwegianBasedContributor(JsonNode element) {
+        return attempt(
+                        () ->
+                                !element.path(Words.AFFILIATIONS)
+                                        .findValues(Words.COUNTRY_CODE)
+                                        .contains(TextNode.valueOf(Words.NO)))
+                .orElse((e) -> Boolean.TRUE);
```

Comment:
Try to avoid using attempt() unless dealing with unknown errors. Here we know the only possible error: the field does not exist

Instead use JsonNode.path() with isMissingNode():

`public abstract JsonNode path(String fieldName) This method is similar to get(String), except that instead of returning null if no such value exists (due to this node not being an object, or object not having value for the specified field), a "missing node" (node that returns true for isMissingNode()) will be returned. This allows for convenient and safe chained access via path calls.`

Reply:
Or: Optional.ofNullable(....get()).orElse() w